game/ConnectionManager.py

```diff
@@ -11,13 +11,17 @@ class ConnectionManager:
     def __init__(self) -> None:
         self.active_connections: dict[Player, WebSocket] = {}
         self.players: dict[str, Player] = {}
+        self.password_dict: dict[str, str] = {}
 
-    async def connect(self, websocket: WebSocket, id: str) -> Player:
+    async def connect(self, websocket: WebSocket, id: str, password) -> Player:
         await websocket.accept()
-        player = self.players.get(id)
-        if not player:
+
+        if self.password_dict.get(id) == password:
+            player = self.players.get(id)
+        else:
             player = Player(id=id)
             self.players[id] = player
+            self.password_dict[id] = password
```

Comment:
not ideal you can just overwrite someones password. if you send the wrong password but his id.